cmd/maya-exporter/app/collector/jivacollector.go

```
@@ -0,0 +1,216 @@
+// method idempotently sends the same descriptors throughout the
+// lifetime of the Collector. If a Collector encounters an error while
+// executing this method, it must send an invalid descriptor (created
+// with NewInvalidDesc) to signal the error to the registry.
+
+// Describe describes all the registered stats metrics from the OpenEBS volumes.
+func (j *JivaStatsExporter) Describe(ch chan<- *prometheus.Desc) {
+	j.Metrics.readIOPS.Describe(ch)
+	j.Metrics.readTimePS.Describe(ch)
+	j.Metrics.readBlockCountPS.Describe(ch)
+	j.Metrics.writeIOPS.Describe(ch)
+	j.Metrics.writeTimePS.Describe(ch)
+	j.Metrics.writeBlockCountPS.Describe(ch)
+	j.Metrics.actualUsed.Describe(ch)
+	j.Metrics.logicalSize.Describe(ch)
+	j.Metrics.sectorSize.Describe(ch)
+	j.Metrics.readLatency.Describe(ch)
+	j.Metrics.writeLatency.Describe(ch)
+	j.Metrics.avgReadBlockCountPS.Describe(ch)
+	j.Metrics.avgWriteBlockCountPS.Describe(ch)
+	j.Metrics.sizeOfVolume.Describe(ch)
+	j.Metrics.volumeUpTime.Describe(ch)
+	j.Metrics.connectionRetryCounter.Describe(ch)
+	j.Metrics.connectionErrorCounter.Describe(ch)
+}
+
+// Collect is called by the Prometheus registry when collecting
+// metrics. The implementation sends each collected metric via the
+// provided channel and returns once the last metric has been sent. The
+// descriptor of each sent metric is one of those returned by
+// Describe. Returned metrics that share the same descriptor must differ
+// in their variable label values. This method may be called
+// way. Bloc	king occurs at the expense of total performance of rendering
+// concurrently and must therefore be implemented in a concurrency safe
+// all registered metrics. Ideally, Collector implementations support
+// concurrent readers.
+
+// Collect collects all the registered stats metrics from the Ope...nEBS volumes.
+// It tries to reconnect with the volume if there is any error via a goroutine.
+func (j *JivaStatsExporter) Collect(ch chan<- prometheus.Metric) {
+	j.collector()
```

We are not making use of the return type i.e. error?
Are we sure of this?

---

Probably `_ = j.collector()` will be good. We can even put some comments as to why we are swallowing the error.
Lets make it a point to mention explicitly if the code needs to swallow error.

---

roger that.